### PR TITLE
Prepare release archive using composer

### LINKED ISSUE #147: Create package script for deploys
To make the deploy easier and prevent mistakes, we could create a script to generate a folder with the built plugin to upload.

Maybe we could do it based on https://github.com/Automattic/sensei/blob/master/gulpfile.js. Or create a new script for that.

### PR DESCRIPTION
**Pull request by merkushin**
Fixes #147 

### Changes proposed in this Pull Request

* Adds `composer.json` config.
* Adds `archive` and `release` scripts for `npm`.
* Adds `.zip` and `/vendor` to `.gitignore`.

### Testing instructions

* Run `composer install` and `npm install`
* Run `npm run release`
* Chech directory for the `syntaxhighlighter.zip`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,9 @@
 		"start": "cgb-scripts start",
 		"build": "cgb-scripts build",
 		"distclean": "rm -rf node_modules && rm -rf dist",
-		"preinstall": "npx npm-force-resolutions"
+		"preinstall": "npx npm-force-resolutions",
+	    "archive": "composer archive --file=$npm_package_name --format=zip",
```
Comment (renatho) [inline package.json]: There is a mix of spaces and tabs here. The same in the `composer.json`.

File: package.json [hunk 2]
```diff
@@ -16,7 +16,9 @@
...
+	    "release": "npm run build && npm run archive"
```
Comment (renatho) [inline package.json]: I wonder if we could add it as part of the `build` script too similar to Sensei, so we keep consistency in the scripts between the projects while releasing. WDYT?

**Review by renatho — COMMENTED** [review]
Awesome! This will be super helpful! 😃
I tested installing in a clean site and it worked well!

I just added some small suggestions.

Also, probably you noticed it only works with node v10 because of the `cgb-scripts` right? Maybe we could take this PR opportunity to also [add it to the `package.json`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines). WDYT?

File: package.json (hunk 2 above)
Comment (merkushin) [reply package.json]: Yes, that's a good idea!

**Review by renatho — APPROVED** [review]
🚀

**Pull request merged**